tests/appsec/waf/test_reports.py

```diff
@@ -4,21 +4,58 @@
 
 from utils import BaseTestCase, context, interfaces, released, bug, irrelevant, missing_feature, flaky, rfc
 import pytest
+import re
 
 
 if context.library == "cpp":
     pytestmark = pytest.mark.skip("not relevant")
 
 
-@missing_feature(library="golang")
+@released(golang="1.38.0")
+@released(java="0.98.0")
 @missing_feature(library="dotnet")
 @missing_feature(library="nodejs")
-@missing_feature(library="java")
 @missing_feature(library="php")
 @missing_feature(library="python")
 @missing_feature(library="ruby")
 class Test_Metrics(BaseTestCase):
-    """ Support In-App WAF metrics report """
+    """ Support In-App WAF monitoring tags and metrics  """
+
+    def test_request_data(self):
+        """ WAF monitoring span tags and metrics are expected to be sent on each request """
+
+        expected_waf_version_tag = "_dd.appsec.event_rules.version"
+        expected_meta_tags = [expected_waf_version_tag]
+        expected_waf_duration_metric = "_dd.appsec.waf.duration"
+        expected_bindings_duration_metric = "_dd.appsec.waf.duration_ext"
```

Comment:
`_dd.appsec.waf.duration_ext` is optional so the test should not fail if it's not present.

Reply:
my bad, thanks. fixing it now